Rework entity generation

Yousei9 opened the pull request:
Seeing as you actively maintain warehousing again I wanted to share some changes I made for myself.

Storehouse icons are changed to a visual style similar to warehouses being less colorful and sporting a dot indicating logistic type.

Generating all entities by hand is prone for typos so I turned generation into a parametrized function duplicating logistic versions from the basic ware- and storehouses.

Anoyomouse commented:
Well, @dgw is doing most of the maintaining these days, but it's still an active mod!

I'm looking at your generation functions and well, they seem pretty similar to what is already being done, except rolling up some things, and when adding being a bit more verbose.

I'll leave it to @dgw if he likes your changes, at the end of the day as long as the tables getting generated are the same then, eh, doesn't matter.

Icons look good, and is probably a good idea as well

dgw commented:
I actually think @Nova-Kast is the best one to weigh in on the new icons, since he's working on graphics tweaks for Factorio 0.17. To my eye, they look too dark, but I actually don't know if the storehouses are part of Nova_Kast's plans. I'll have to ping him.

The code diff is pretty overwhelming to review on my phone, so I'll put that off for now. :grinning:

dgw commented on prototypes/entity.lua:
```diff
@@ -1,302 +1,148 @@
+local warehouse_active_provider = createLogisticContainer("warehouse", "active-provider")
+local warehouse_passive_provider = createLogisticContainer("warehouse", "passive-provider")
+local warehouse_storage = createLogisticContainer("warehouse", "storage")
+warehouse_storage.inventory_size = 1000
```
Storage warehouse has 2000 slots.

Storage logistic variants should have one logistic slot to allow filtering.
```suggestion
warehouse_storage.inventory_size = 2000
warehouse_storage.logistic_slots_count = 1
```

dgw commented on prototypes/entity.lua:
```diff
@@ -1,302 +1,148 @@
+		type = "container",
+		name = "warehouse-basic",
+		icon = "__Warehousing__/graphics/icons/warehouse-basic.png",
+    icon_size = 32,
```
Wrong indentation.

dgw commented on prototypes/entity.lua:
```diff
@@ -1,302 +1,148 @@
+local storehouse_active_provider = createLogisticContainer("storehouse", "active-provider")
+local storehouse_passive_provider = createLogisticContainer("storehouse", "passive-provider")
+local storehouse_storage = createLogisticContainer("storehouse", "storage")
+storehouse_storage.inventory_size = 300
```
Storage logistic variants should have one logistic slot to allow filtering.
```suggestion
storehouse_storage.inventory_size = 300
storehouse_storage.logistic_slots_count = 1
```

Yousei9 replied:
Yes, that pull was made before filter slots where a thing.

dgw commented on prototypes/entity.lua:
```diff
@@ -1,302 +1,148 @@
+		type = "container",
+		name = "storehouse-basic",
+		icon = "__Warehousing__/graphics/icons/storehouse-basic.png",
+    icon_size = 32,
```
Wrong indentation.

dgw commented on prototypes/entity.lua:
```diff
@@ -1,302 +1,148 @@
-	define_storehouse("buffer", "buffer"),
-})
-
+--[[ Copyright (c) Optera 2017
```
It would be nice to leave @Anoyomouse's copyright line in here, and simply _add_ yours instead of replacing it, but I'm OK with this if he is.

dgw commented on prototypes/entity.lua:
```diff
@@ -1,302 +1,148 @@
+--[[ Copyright (c) Optera 2017
+ * Part of the warehousing mod
+ *
+ * See LICENCE in the project directory for licence information
```
This is old. The current license header line refers to a different file.
```suggestion
 * See License.txt in the project directory for license information.
```

dgw commented on prototypes/entity.lua:
```diff
@@ -1,302 +1,148 @@
-})
-
+--[[ Copyright (c) Optera 2017
+ * Part of the warehousing mod
```
Capitalization nitpick.
```suggestion
 * Part of the Warehousing mod
```

dgw commented on prototypes/entity.lua:
```diff
@@ -1,302 +1,148 @@
+  warehouse_passive_provider,
+  warehouse_storage,
+  warehouse_buffer,
+  warehouse_requester
```
I'm not that well-versed in Lua, but I believe trailing commas are allowed. Using one here would be more future-proof against adding/removing warehouse types. (Just in terms of diff cleanliness, of course.)
```suggestion
  warehouse_requester,
```

Yousei9 replied:
Lua does allow trailing comma on lists, I use them myself a fair bit whenever I'm uncertain if I need to add more.

Yousei9 commented:
Merging such old pull requests always is a chore. I hope I got all of those tiny change requests.

dgw commented:
I really appreciate the quick turnaround, @Yousei9!

Assuming I can get my Lua interpreter going, I'll try to verify that the old and new code produce equivalent output without actually loading the mod (because I'm much too lazy to install Factorio and start a test game just for this :grin:).

If all is well, would you prefer that I handle squashing the tweaks you added today into the appropriate commit, or do you want to do that yourself? I know this is an old PR, so I'm happy to do the rebase for you if you'd rather move on to other things.

Yousei9 commented:
I totally overlooked "-shadow" had been added to warehouse sprites. Feels very inconsistent to me. To reduce overall mod filesize you may want to break apart sprite and shadow like all base factorio entities do.
Removed the -shadow from the sprites so those names can safely be used for shadow sprite.

I very much prefer unscaled icons myself. I'll add a mod option to toggle it on/off defaulting to on.

I don't recall why I removed hardness and changed health, but neither of them really impact gameplay in a huge way. If you want I'll add health as parameter to the creation function.

scale_entity_info_icon is a property of crafting machine not chests
https://wiki.factorio.com/Prototype/CraftingMachine

dgw commented:
The `-shadow` suffix on sprite filenames was left over from… I have no idea when. It was there when I got involved with the mod about a year ago, and had probably been there for ages already. Neither I nor @Anoyomouse can regenerate the sprites from the source models (we don't have them), so any thoughts of splitting the graphics into separate files will have to wait for @Nova-Kast to come out of hibernation.

I do like that this rewrite got rid of a bunch of dead code (the circuit-connector functions) that was no longer used. That's a big plus.

The new setting… Honestly, adding a setting is just out of scope for this PR, which is to rewrite the entity-generation code and change a few icons. The icon updates, even, should probably be in a separate PR, but I'm not too bothered. It kind of goes with the rewrite… if you squint.

Do feel free to implement the setting in a separate PR, though. I'm just saying "no" to including it with this code rewrite, because adding a new setting is its own thing and should be handled as such.

Yousei9 commented:
The setting needs to be added into entity generation. A new pull request would need to be based on this rewrite, but I have no idea how to make that work in github. Cramming that into here seemed not too bad to me.

dgw commented:
Two options, both starting with removing the added setting from this PR:

1. Wait until this PR is merged, then open a new one with the setting addition.
2. Open a new PR with the setting addition on top of this PR, then deal with any rebasing needed after this one is merged.

Speaking of rebasing, are you up for doing that to this PR? Or should I put together a cleaned-up history? I noticed you're using "Add files via upload", and rebasing can't be done through GitHub's web UI.

Yousei9 commented:
Since I can't do a rebase you'll have to do the cleanup and merge.

dgw commented:
Squashed the code commits down, and removed the icon changes to a separate PR (#56) while I was at it.

This should be 100% good to go once I have a chance to fire up a quick test game of Factorio with this version of the mod to make sure I didn't break anything in the rebase.

Yousei9 commented:
The lines setting scale_info_icons = true are missing.
Either I uploaded the wrong file or the cleanup went wrong.

dgw commented:
They're in the basic entity definitions, below the `picture` attributes.

dgw commented:
Right-o, this tested out great.
